### PR TITLE
Update gitignore to other comparable ROCm projects

### PR DESCRIPTION
**Pull request by CharlieL7**
Small change to .gitignore to have it ignore compiled files. Should help prevent the addition of compiled files to the repository.

### CODE REVIEW

**Comment by kahmed10** [comment]
I am okay with updating our .gitignore file to be more generic and in line with other ROCmSoftwarePlatform .gitignore files. Maybe you can take their examples and add more to the ignore list?
https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/miopen-dialect/.gitignore
https://github.com/ROCmSoftwarePlatform/pytorch/blob/master/.gitignore
https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/develop/.gitignore

It doesn't have to be super exhaustive since not all of the technologies in the other gitignore files are relevant to MIGraphX. I would at least include more of the code editors such as vscode and other swap temporary extensions.

File: .gitignore [hunk 1]
```diff
@@ -1 +1,66 @@
...
+/GTAGS
+/ID
+.gitusers
+/compile_commands.json
+
+/CMakeSettings.json
+
+# Visual studio files
+!/.vscode/extensions.json
+!/.vscode/settings_recommended.json
```
Comment (pfultz2) [inline .gitignore:53]: What does the `!` do here?

Comment (CharlieL7) [reply .gitignore:53]: ! negates the next pattern. I think these lines would actually include these files explicitly. I'll remove them for now.

**Review by pfultz2 — APPROVED** [review]
Are you going to remove the .pyc files already committed at some point?

**Review by turneram — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -1 +1,63 @@
...
+/compile_commands.json
+
+/CMakeSettings.json
+
+
+#==============================================================================#
+# Directories to ignore (do not add trailing '/'s, they skip symlinks).
+#==============================================================================#
+# Nested build directory
+/build*
```
Comment (turneram) [inline .gitignore:56]: I think `/doc*` would also be good to add.

File: .gitignore [hunk 3]
```diff
@@ -1 +1,63 @@
...
+
```
Comment (turneram) [inline .gitignore:57]: I think it would also be good to add `/doc/html*`

**Pull request merged**